HW06.Rmd

```diff
@@ -61,28 +91,58 @@ detach(package:EnsDb.Hsapiens.v75, unload = T)
 detach(package:ensembldb, unload = T)
 
 #add in your own gene of interest!!! 
-genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "CICP27", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "DDX11L1", "WBP1LP7", "SAMD11")
 
-
-GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest) %>% 
+# gather the data frame and tidy
+  gather(key = "cell_line", value = "TPM", 2:46) %>% 
+  separate(col = cell_line, into = c("1", "cell_line", "No."), sep = "\\.") %>% 
+  dplyr::select("gene_symbol", "cell_line", "TPM")
 
 #Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
 
-#print out the t-test results
+p <- c()
+i = 0
+for (x in genes_of_interest) {
+# filter data for certain gene
+  i = i + 1
+y <- GBM.genes.of.interest %>% 
+  filter(gene_symbol == x)
+#print out the t-test results as a table
+p[i] <- t.test(y$TPM[y$cell_line == "H3K27_Mutated"], y$TPM[y$cell_line == "WT"])$p.value
+}
+t_test <- data.frame(genes_of_interest, p)
+knitr::kable(t_test, format = "markdown", align = "c", col.names = c("Genes", "p-value"))
 ```
 
 
 
 Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
 
 ```{r create a graphing function}
-#to work in the tidyverse, it will be easier to make tidy the dataframe first
-
-
-
 #create a graphing function
+graph <- function(x) {
+# make the boxplot graph
+  ggplot(x, aes(x = cell_line, y = TPM)) +
+  geom_boxplot(aes(fill = cell_line)) +
+# add labels
+  labs(x = "H3K27",
+         y = "Expression_TPM") +
+# adjust theme
+  theme_bw() +
+  theme(legend.position = "none",
+        plot.title = element_text(hjust = 0.5))
+}
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest
+for (x in genes_of_interest) {
+  image <- GBM.genes.of.interest %>% 
+    filter(gene_symbol == x) %>% 
+    graph() +
+# add title
+    labs(title = paste(x, "Expression in GBM models\nby H3K27 Mutated or WT Status"))
+  print(image)
+}

```

Comment:
This is where I wish I had categorized the genetic background early on. I was struggling to distinguish between WT and H3K27M for generating the graphs, but it seems like you found a great way to do that. Also, neat idea to define a function for generating a graph, and then applying that function in a for loop to generate a boxplot for each gene. Overall, excellent way to generate all the plots!